src/pages/ValidateLoginPage/index.website.tsx

```diff
@@ -27,6 +28,14 @@ function ValidateLoginPage({
     const isUserClickedSignIn = !login && isSignedIn && (autoAuthState === CONST.AUTO_AUTH_STATE.SIGNING_IN || autoAuthState === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN);
     const shouldStartSignInWithValidateCode = !isUserClickedSignIn && !isSignedIn && (!!login || !!exitTo);
 
+    // If the magic link was previously expired (FAILED), we want to clear the failed state so the magic code expired won't show briefly.
+    if (firstRenderRef.current) {
+        firstRenderRef.current = false;
+        if (autoAuthState === CONST.AUTO_AUTH_STATE.FAILED) {
+            Session.clearAutoAuthState();
+        }
+    }
```

Comment:
The solution here is a bit different from the proposal. Instead of clearing the autoAuthState when pressing "request one here" text button, I clear it when the page function is called for the first time.

I do this because on Desktop, there is no "request one here" text button, similar to native. Instead, the ValidateLoginPage is closed. So, the issue still happens if we:

1. Open an expired magic link
2. User is redirected back to the sign-in page with a form error shown
3. Open the new magic link
4. Magic code expired is shown briefly

I tried clearing it in an `useEffect`, but the magic code expired is still shown briefly.